lib/vector-core/Cargo.toml

```diff
@@ -22,6 +22,7 @@ http = { version = "0.2.4", default-features = false }
 hyper-proxy = { version = "0.9.1", default-features = false, features = ["openssl-tls"] }
 # pinned due to https://github.com/tkaitchuck/aHash/issues/95
 indexmap = { version = "~1.6.2", default-features = false, features = ["serde"] }
+internal_event = { path = "internal-event", default-features = false }
```

Comment:
It's not clear to me why you'd want this to be a sub-crate, rather than just an exported trait from the top-level vector-core? Without a specific reason for making such a hard separation I'd argue we're better off having `vector_core::internal_event` be a module. 

Reasons you might want a sub-crate: 

* You are working around build-times and need to have a separate crate that is independently compiled, testable. 
* You believe the code might be a candidate for extraction from the repo entirely.
* You intend to force the crate to be generic, not coupled to one of Vector's key types. 

The first and last point are why `buffers` is a sub-crate here, as an example.

Reply:
Ah, I'd made it a sub-crate because I planned for `buffers` to take a dependency on it to instrument the code in `buffers/src/lib.rs`. My current understanding is that if `internal_event` is instead a module in `vector_core` then `buffers` will need to depend on `vector_core` which should be avoided. I may be missing some nuances here.

Reply:
Ah ha, that would do it. Makes sense.